docs/changes/coordinates/16246.feature.rst

```diff
@@ -0,0 +1,7 @@
+The ``SkyCoord`` and ``BaseCoordinateFrame`` ``separation()`` methods now
+accept an optional ``frame_mismatch`` argument.
+The ``"warn"`` value (default) makes the method emit a warning if computing the
+separation requires a coordinate transformation that is not a simple rotation.
```

Comment:
Add here something like "simple rotation, so that the separation between coordinates may depend on the choice of which coordinate frame to measure the separation in."